### PR TITLE
Choose board size

### PR DESCRIPTION
**Pull request by drewb8L**
This is a draft PR, the ability to have different size boards is implemented but no UI exists to change it. Error handling and user selection to be enabled soon.

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -5,11 +5,18 @@
...
-    expect(subject.board_spaces).to eq({ '7': ' ', '8': ' ', '9': ' ', '4': ' ', '5': ' ', '6': ' ', '1': ' ',
-                                         '2': ' ', '3': ' ' })
+    expect(subject.board_spaces).to eq({ 0 => '00', 1 => '01', 2 => '02', 3 => '03', 4 => '04', 5 => '05', 6 => '06',
+                                         7 => '07', 8 => '08' })
   end
 
   it 'square 7 should be string with space " "' do
-    expect(subject.board_spaces[:'7']).to eq(' ')
+    expect(subject.board_spaces[7]).to eq('07')
+  end
```
Comment (digitopolis) [inline spec/board_spec.rb]: Looks like the description of the test needs to be updated -> 'square 7 should be string "07"'

File: spec/board_spec.rb [hunk 2]
```diff
@@ -5,11 +5,18 @@
...
+
+  context Board do
+    let(:board) { described_class.new(5) }
+    it 'should have a size of 9' do
+      expect(board.board_spaces.length).to be(25)
+    end
```
Comment (digitopolis) [inline spec/board_spec.rb]: This test is confusing - would like to see more descriptive `context`, something like 'initialized with row length 5'

and then
```
it 'should have a size of 25' do
```

File: src/ttt_board.rb [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
-  end
-
+  # def draw_board
+  #   puts " #{@board_spaces[:'7']} |#{@board_spaces[:'8']} |#{@board_spaces[:'9']}"
+  #   puts ' --+--+--'
+  #   puts " #{@board_spaces[:'4']} |#{@board_spaces[:'5']} |#{@board_spaces[:'6']}"
+  #   puts ' --+--+--'
+  #   puts " #{@board_spaces[:'1']} |#{@board_spaces[:'2']} |#{@board_spaces[:'3']}"
+  # end
+  #
```
Comment (digitopolis) [inline src/ttt_board.rb]: Did you mean to leave these commented out boards in here?

File: src/board.rb [hunk 4]
```diff
@@ -2,7 +2,50 @@
...
+    size.times { |element| board.store(element, format('%02d', element)) }
+    board
+  end
+
+  def make_example_board
+    board = {}
+    size = @rows * @rows
+    size.times { |element| board.store(element, format('%02d', element)) }
+    board
+  end
```
Comment (digitopolis) [inline src/board.rb:25]: These functions are identical – what are the different use cases for them?

**Review by digitopolis — COMMENTED** [review]
Looks pretty good! I haven't tried it out locally yet, but I like how you handle creating the board programmatically now 👍 

Just left a couple comments. Whenever you're done working on a branch and want a review, go ahead and take it out of Draft state ✅

File: src/board.rb (hunk 4 above)
Comment (drewb8L) [reply src/board.rb:25]: Make example board os the function that displays the board with the numbers at the beginning of the game.  My intent was to have one function display an example ant the other display a blank board, but I don't think that's necessary any more so I'll take one of them out.

File: src/ttt_board.rb (hunk 3 above)
Comment (drewb8L) [reply src/ttt_board.rb]: the board parent class has the methods to draw the board now, so these can be deleted.

File: spec/board_spec.rb (hunk 1 above)
Comment (drewb8L) [reply spec/board_spec.rb]: I'll fix this

File: spec/board_spec.rb (hunk 2 above)
Comment (drewb8L) [reply spec/board_spec.rb]: Yes, This should be 25, fixing.

**Pull request closed without merging**